wandb/sdk/launch/_project_spec.py

```diff
@@ -175,24 +173,22 @@ def image_name(self) -> str:
             assert self.job is not None
             return wandb.util.make_docker_image_name_safe(self.job.split(":")[0])
 
-    def _initialize_image_job_tag(self) -> Optional[str]:
-        if self.job is not None:
-            job_name, alias = self.job.split(":")
-            # Alias is used to differentiate images between jobs of the same sequence
-            _image_tag = f"{alias}-{job_name}"
-            _logger.debug(f"{LOG_PREFIX}Setting image tag {_image_tag}")
-            return wandb.util.make_docker_image_name_safe(_image_tag)
-        return None
-
-    @property
-    def image_uri(self) -> str:
-        if self.docker_image:
-            return self.docker_image
-        return f"{self.image_name}:{self.image_tag}"
-
-    @property
-    def image_tag(self) -> str:
-        return self._image_tag[:IMAGE_TAG_MAX_LENGTH]
+    def build_required(self) -> bool:
+        """Checks the source to see if a build is required."""
+        if self.source == LaunchSource.JOB:
+            assert isinstance(self.job, str)
+            job_info = self.job.split(":")
+            if len(job_info) != 2:
+                raise LaunchError(
+                    "Job must be in the format <entity>/<propject>/<collection>:<alias> or <entity>/<propject>/<collection>:v<version>"
+                )
+            alias = job_info[1]
+            # a build is only not required if the user has specified
+            # a version index, since all other aliases can
+            # be moved from version to version
+            return re.match(r"v\d+", alias) is None
+        else:
+            return True
```

Comment:
```suggestion
        if self.source != LaunchSource.JOB:
            return true
            
       assert isinstance(self.job, str)
       job_info = self.job.split(":")
       if len(job_info) != 2:
           raise LaunchError(
               "Job must be in the format <entity>/<propject>/<collection>:<alias> or <entity>/<propject>/<collection>:v<version>"
           )
       alias = job_info[1]
       # a build is only not required if the user has specified
       # a version index, since all other aliases can
       # be moved from version to version
       return re.match(r"v\d+", alias) is None
```

Reply:
+1 on this change